spyder/plugins/completion/kite/utils/install.py

```diff
@@ -0,0 +1,186 @@
+# -*- coding: utf-8 -*-
+
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+# (see spyder/__init__.py for details)
+
+"""Kite installation functions."""
+
+# Standard library imports
+from __future__ import print_function
+import os
+import os.path as osp
+import re
+import subprocess
+import sys
+from tempfile import gettempdir
+from urllib.request import urlretrieve
+
+# Third party imports
+from qtpy.QtCore import QThread, Signal
+
+# Local imports
+from spyder.py3compat import to_text_string
+from spyder.plugins.completion.kite.utils.status import check_if_kite_installed
+
+
+class KiteInstallationThread(QThread):
+    """Thread to handle the installation process of kite."""
+
+    # Installer URLs
+    WINDOWS_URL = "https://release.kite.com/dls/windows/current"
+    LINUX_URL = "https://release.kite.com/dls/linux/current"
+    MAC_URL = "https://release.kite.com/dls/mac/current"
+
+    # Process status
+    NO_STATUS = "No status"
+    DOWNLOADING_SCRIPT = "Downloading Kite script installer"
+    DOWNLOADING_INSTALLER = "Downloading Kite installer"
+    INSTALLING = "Installing Kite"
+    FINISHED = "Install finished"
+    ERRORED = "Error"
```

Comment:
@dalthviz remember to add the translation func.

`NO_STATUS = "No status"` -> ` NO_STATUS = _("No status")`